### PR TITLE
scipy.eigsh: set linear solver

### PR DESCRIPTION
**Pull request by vlukes**
Setting up a linear solver for matrix factorization can rapidly speed up the computation of eigenvalues:

- eigsh from Scipy : 2648s
- Octave: 214s
- Matlab: 224s
- **eigsh + cholesky: 62s**

See discussion under #1105.

### CODE REVIEW

File: sfepy/examples/linear_elasticity/modal_analysis_declarative.py [hunk 1]
```diff
@@ -107,6 +107,10 @@ def define(n_eigs=6, approx_order=1, density=7850., young=210e9, poisson=0.3):
             'which': 'LM',
             'sigma': 0,
             'tol': 1e-8,
+            'linear_solver': 'splu',
+            # 'cholesky' solver is usually faster, but
+            # scikit-sparse package is reuired
```
Comment (rc) [inline sfepy/examples/linear_elasticity/modal_analysis_declarative.py:112]: reuired -> required

File: sfepy/solvers/eigen.py [hunk 2]
```diff
@@ -76,6 +76,9 @@ class ScipyEigenvalueSolver(EigenvalueSolver):
             see :func:`scipy.sparse.linalg.eigs()`
             or :func:`scipy.sparse.linalg.eigsh()`. For dense problmes,
             only 'LM' and 'SM' can be used"""),
+        ('linear_solver', "{'cholesky', 'splu'}", None, False,
+         """The method used to construct a inverse linear operator. If None, the
```
Comment (rc) [inline sfepy/solvers/eigen.py:80]: an inverse

**Comment by rc** [comment]
Nice, thanks! Maybe use it also in the `sfepy/examples/linear_elasticity/modal_analysis.py` example?

So in principle, one could use also MUMPS or UMFPACK as the internal solver, right?

File: sfepy/examples/linear_elasticity/modal_analysis_declarative.py [hunk 3]
```diff
@@ -107,6 +107,10 @@ def define(n_eigs=6, approx_order=1, density=7850., young=210e9, poisson=0.3):
             'which': 'LM',
             'sigma': 0,
             'tol': 1e-8,
+            'linear_solver': ('ls.mumps', {}),
+            # 'cholesky' solver is usually faster, but
```
Comment (rc) [inline sfepy/examples/linear_elasticity/modal_analysis_declarative.py:111]: Is the comment still valid?

File: sfepy/examples/linear_elasticity/modal_analysis_declarative.py [hunk 4]
```diff
@@ -107,6 +107,10 @@ def define(n_eigs=6, approx_order=1, density=7850., young=210e9, poisson=0.3):
             'which': 'LM',
             'sigma': 0,
             'tol': 1e-8,
+            'linear_solver': ('ls.mumps', {}),
```
Comment (rc) [inline sfepy/examples/linear_elasticity/modal_analysis_declarative.py:110]: I would set the solver to something that is not optional dependency. MUMPS libraries need not be available.

Comment (vlukes) [reply sfepy/examples/linear_elasticity/modal_analysis_declarative.py]: The default linear solver is set to `ls.scipy_superlu` and the comment is updated.

**Review by rc — APPROVED** [review]

**Pull request merged**